src/loader.js

```diff
@@ -0,0 +1,91 @@
+let assume = require('assume');
+let debug = require('debug')('taskcluster:loader');
+let Promise = require('promise');
+
+function loader(componentDirectory, requiredVirtualComponents) {
+  /**
+   * If present, this is a list of component names that a loader
+   * must define.  These are likely things like configuration values
+   */
+  if (requiredVirtualComponents) {
+    assume(requiredVirtualComponents).is.array();
+  } else {
+    requiredVirtualComponents = [];
+  }
```

Comment:
To minimize dependencies I would be tempting to do:
```js
assert(requiredVirtualComponents.every(e => typeof(e) === 'string'),
          "requiredVirtualComponents must be an array of strings");
```

But I like that you picked up `assume` :rocket:, it's really nice... I've never felt save with `.should`.